tests/lookup/tests.py

```diff
@@ -909,3 +911,16 @@ def test_custom_lookup_none_rhs(self):
         field = query.model._meta.get_field('nulled_text_field')
         self.assertIsInstance(query.build_lookup(['isnull_none_rhs'], field, None), IsNullWithNoneAsRHS)
         self.assertTrue(Season.objects.filter(pk=season.pk, nulled_text_field__isnull_none_rhs=True))
+
+    def test_outerref_in_dates(self):
+        dates = Author.objects.filter(date__year__gte=OuterRef("year"))
+        dates_subq = Subquery(dates.values("id"))
+        years = Article.objects.filter(author__in=dates_subq)
+        self.assertQuerysetEqual(
+            years,
+            ['<Article: Article 1>', '<Article: Article 2>',
```

Comment:
Use the indentation style of the other tests. Actually, you can use `assertCountEqual()` instead. Maybe it makes sense to create another Article with a different year to test that the results are correct and not just returning everything.